client-go/Makefile

```diff
@@ -0,0 +1,51 @@
+include ../includes.mk
+
+# the filepath to this repository, relative to $GOPATH/src
+repo_path = github.com/deis/deis/client-go
+
+GO_FILES = $(wildcard *.go)
+GO_PACKAGES = parser cmd controller/api controller/client controller/models/users
+GO_PACKAGES_REPO_PATH = $(addprefix $(repo_path)/,$(GO_PACKAGES))
+
+COMPONENT = $(notdir $(repo_path))
+IMAGE =  $(IMAGE_PREFIX)/$(COMPONENT):$(BUILD_TAG)
+
+build:
+	CGO_ENABLED=0 godep go build -a -installsuffix cgo -ldflags '-s' -o deis .
+
+installer:
+	rm -rf dist && mkdir -p dist
+	CGO_ENABLED=0 godep go build -a -installsuffix cgo -ldflags '-s' -o dist/deis .
+	@if [ ! -d makeself ]; then git clone -b single-binary https://github.com/deis/makeself.git; fi
+	PATH=./makeself:$$PATH BINARY=deis makeself.sh --bzip2 --current --nox11 dist \
+		dist/deis-`cat deis-version`-`go env GOOS`-`go env GOARCH`.run "Deis Client" \
+		echo 'deis is in the current directory' \
+		&& echo 'Please move deis to a directory in your search PATH.' \
+		&& echo \
+		&& echo 'See http://docs.deis.io/ for documentation.' \
+		&& echo"
+
+install:
+	godep go install -v .
+
+setup-root-gotools:
+	sudo GOPATH=/tmp/tmpGOPATH go get -u -v golang.org/x/tools/cmd/cover
```

Comment:
This is a voodoo incantation necessary to get `go vet` installed on some platforms, like my Mac. We tried to remove it once and found yes, it still causes errors to install `go vet` as other than root.